path/spanning_tree.go

```diff
@@ -5,101 +5,166 @@
 package path
 
 import (
+	"container/heap"
+	"math"
 	"sort"
 
 	"github.com/gonum/graph"
 	"github.com/gonum/graph/concrete"
-	"github.com/gonum/graph/internal"
 )
 
-// EdgeListerGraph is an undirected graph than returns its complete set of edges.
-type EdgeListerGraph interface {
+// UndirectedWeighter is an undirected graph that returns distinct edge weights.
```

Comment:
This is the terminology used in the algorithm proofs. Distinct means there are no repeated weights in the set of E. The consequence of this invariant being broken is not terrible (the algorithms find *an* MST instead of *the* MST), so I don't check for it - the user should know whether they care about the consequence and what it means.